### PR TITLE
Apply "readability-implicit-bool-conversion" clang-tidy check

### PR DESCRIPTION
**Pull request by florian-kuebler**
Test: Compile

### CODE REVIEW

File: src/OrbitGl/PrimitiveAssembler.cpp [hunk 1]
```diff
@@ -340,7 +340,7 @@ void PrimitiveAssembler::StartNewFrame() { batcher_->ResetElements(); }
 const orbit_client_protos::TimerInfo* PrimitiveAssembler::GetTimerInfo(PickingId id) const {
   const PickingUserData* data = GetUserData(id);
 
-  if (data && data->timer_info_) {
+  if ((data != nullptr) && (data->timer_info_ != nullptr)) {
```
Comment (dpallotti) [inline src/OrbitGl/PrimitiveAssembler.cpp:343]: I don't mind them, but personally  I wouldn't have added the parentheses here and in some other places in this change where you added `!= nullptr`. I think it's normal to write `ptr != nullptr && ptr->foo`.

File: src/OrbitQt/CallTreeWidget.cpp [hunk 2]
```diff
@@ -919,7 +919,7 @@ void CallTreeWidget::ProgressBarItemDelegate::paint(QPainter* painter,
   }
 
   bool highlight = index.data(CallTreeViewItemModel::kMatchesCustomFilterRole).toBool();
-  if (option.state & QStyle::State_Selected) {
+  if (option.state & QStyle::State_Selected != 0u) {
```
Comment (dpallotti) [inline src/OrbitQt/CallTreeWidget.cpp:922]: Doesn't `!=` have higher precedence than `&`? We need parentheses here.

File: src/OrbitQt/include/OrbitQt/CustomSignalsTreeView.h [hunk 3]
```diff
@@ -25,7 +25,7 @@ class CustomSignalsTreeView : public QTreeView {
 
   void mousePressEvent(QMouseEvent* event) override {
     QModelIndex index = indexAt(event->pos());
-    bool alt_pressed = event->modifiers() & Qt::AltModifier;
+    bool alt_pressed = event->modifiers() & Qt::AltModifier != 0u;
```
Comment (dpallotti) [inline src/OrbitQt/include/OrbitQt/CustomSignalsTreeView.h:28]: And here.

File: src/OrbitQt/orbitglwidget.cpp [hunk 4]
```diff
@@ -153,9 +154,9 @@ void OrbitGLWidget::leaveEvent(QEvent*) { gl_canvas_->SetIsMouseOver(false); }
 
 void OrbitGLWidget::keyPressEvent(QKeyEvent* event) {
   if (gl_canvas_) {
-    bool ctrl = event->modifiers() & Qt::ControlModifier;
-    bool shift = event->modifiers() & Qt::ShiftModifier;
-    bool alt = event->modifiers() & Qt::AltModifier;
+    bool ctrl = event->modifiers() & Qt::ControlModifier != 0u;
+    bool shift = event->modifiers() & Qt::ShiftModifier != 0u;
+    bool alt = event->modifiers() & Qt::AltModifier != 0u;
```
Comment (dpallotti) [inline src/OrbitQt/orbitglwidget.cpp:159]: And here.

File: src/OrbitQt/orbitglwidget.cpp [hunk 5]
```diff
@@ -140,8 +140,9 @@ void OrbitGLWidget::mouseDoubleClickEvent(QMouseEvent* event) {
 
 void OrbitGLWidget::mouseMoveEvent(QMouseEvent* event) {
   if (gl_canvas_) {
-    gl_canvas_->MouseMoved(event->x(), event->y(), event->buttons() & Qt::LeftButton,
-                           event->buttons() & Qt::RightButton, event->buttons() & Qt::MiddleButton);
+    gl_canvas_->MouseMoved(event->x(), event->y(), event->buttons() & Qt::LeftButton != 0u,
+                           event->buttons() & Qt::RightButton != 0u,
+                           event->buttons() & Qt::MiddleButton != 0u);
```
Comment (dpallotti) [inline src/OrbitQt/orbitglwidget.cpp:145]: And here.

File: src/OrbitQt/orbitglwidget.cpp [hunk 6]
```diff
@@ -169,9 +170,9 @@ void OrbitGLWidget::keyPressEvent(QKeyEvent* event) {
 
 void OrbitGLWidget::keyReleaseEvent(QKeyEvent* event) {
   if (gl_canvas_) {
-    bool ctrl = event->modifiers() & Qt::ControlModifier;
-    bool shift = event->modifiers() & Qt::ShiftModifier;
-    bool alt = event->modifiers() & Qt::AltModifier;
+    bool ctrl = event->modifiers() & Qt::ControlModifier != 0u;
+    bool shift = event->modifiers() & Qt::ShiftModifier != 0u;
+    bool alt = event->modifiers() & Qt::AltModifier != 0u;
```
Comment (dpallotti) [inline src/OrbitQt/orbitglwidget.cpp:175]: And here.

File: src/OrbitQt/orbitglwidget.cpp [hunk 7]
```diff
@@ -182,10 +183,10 @@ void OrbitGLWidget::wheelEvent(QWheelEvent* event) {
   if (gl_canvas_) {
     if (event->orientation() == Qt::Vertical) {
       gl_canvas_->MouseWheelMoved(event->x(), event->y(), event->delta() / 8,
-                                  event->modifiers() & Qt::ControlModifier);
+                                  event->modifiers() & Qt::ControlModifier != 0u);
     } else {
       gl_canvas_->MouseWheelMovedHorizontally(event->x(), event->y(), event->delta() / 8,
-                                              event->modifiers() & Qt::ControlModifier);
+                                              event->modifiers() & Qt::ControlModifier != 0u);
```
Comment (dpallotti) [inline src/OrbitQt/orbitglwidget.cpp:189]: And here.

**Review by dpallotti — COMMENTED** [review]
This is my favorite clang-tidy check!

But careful: I think you forgot some parentheses where necessary with `&`.

File: src/OrbitQt/CallTreeWidget.cpp (hunk 2 above)
Comment (dpallotti) [reply src/OrbitQt/CallTreeWidget.cpp:922]: Oh nice, this actually triggers a compiler error

Comment (florian-kuebler) [reply src/OrbitQt/CallTreeWidget.cpp:922]: Thanks for the catch (while I am happy that the compiler also got this). The changes were autogenerated by clang-tidy, and I missed that when proofreading them.

File: src/OrbitGl/PrimitiveAssembler.cpp (hunk 1 above)
Comment (florian-kuebler) [reply src/OrbitGl/PrimitiveAssembler.cpp]: Removed it.

File: src/OrbitQt/CallTreeWidget.cpp (hunk 2 above)
Comment (florian-kuebler) [reply src/OrbitQt/CallTreeWidget.cpp]: Done.

File: src/OrbitQt/include/OrbitQt/CustomSignalsTreeView.h (hunk 3 above)
Comment (florian-kuebler) [reply src/OrbitQt/include/OrbitQt/CustomSignalsTreeView.h]: Done.

File: src/OrbitQt/orbitglwidget.cpp (hunk 5 above)
Comment (florian-kuebler) [reply src/OrbitQt/orbitglwidget.cpp]: Done.

File: src/OrbitQt/orbitglwidget.cpp (hunk 4 above)
Comment (florian-kuebler) [reply src/OrbitQt/orbitglwidget.cpp]: Done.

File: src/OrbitQt/orbitglwidget.cpp (hunk 6 above)
Comment (florian-kuebler) [reply src/OrbitQt/orbitglwidget.cpp]: Done.

File: src/OrbitQt/orbitglwidget.cpp (hunk 7 above)
Comment (florian-kuebler) [reply src/OrbitQt/orbitglwidget.cpp]: Done.

File: src/OrbitQt/orbittreeview.cpp [hunk 8]
```diff
@@ -174,7 +174,7 @@ void OrbitTreeView::Refresh(RefreshMode refresh_mode) {
 
 void OrbitTreeView::resizeEvent(QResizeEvent* event) {
   const bool width_resized = event->size().width() != event->oldSize().width();
-  if (width_resized && model_ != nullptr && model_->GetDataView()) {
+  if (width_resized && model_ != nullptr && (model_->GetDataView() != nullptr)) {
```
Comment (dpallotti) [inline src/OrbitQt/orbittreeview.cpp:177]:
```suggestion
  if (width_resized && model_ != nullptr && model_->GetDataView() != nullptr) {
```
Just for consistency with the rest of the condition.

**Review by dpallotti — APPROVED** [review]

File: src/OrbitQt/orbittreeview.cpp (hunk 8 above)
Comment (florian-kuebler) [reply src/OrbitQt/orbittreeview.cpp]: Done.

**Pull request merged**